plugins/example-code-snippets/src/inject-example-code-snippets.ts

```diff
@@ -1,140 +1,138 @@
-  rust={\`${snippetForLanguage(ExampleLanguage.RUST, snippetId)}\`}
-  ruby={\`${snippetForLanguage(ExampleLanguage.RUBY, snippetId)}\`}
-  elixir={\`${snippetForLanguage(ExampleLanguage.ELIXIR, snippetId)}\`}
-  swift={\`${snippetForLanguage(ExampleLanguage.SWIFT, snippetId)}\`}
-  dart={\`${snippetForLanguage(ExampleLanguage.DART, snippetId)}\`}
-  cli={\`${snippetForLanguage(ExampleLanguage.CLI, snippetId)}\`}
-/>
-`;
-}
-
-function replaceValueWithExampleFileTabs(literal: unist.Literal): void {
-  const value = literal.value as string;
-  literal.value = `<SdkExampleFileTabsImpl
-    js={\`${fileTabContentsForLanguage(value, ExampleLanguage.JAVASCRIPT)}\`}
-    python={\`${fileTabContentsForLanguage(value, ExampleLanguage.PYTHON)}\`}
-    java={\`${fileTabContentsForLanguage(value, ExampleLanguage.JAVA)}\`}
-    kotlin={\`${fileTabContentsForLanguage(value, ExampleLanguage.KOTLIN)}\`}
-    go={\`${fileTabContentsForLanguage(value, ExampleLanguage.GO)}\`}
-    csharp={\`${fileTabContentsForLanguage(value, ExampleLanguage.CSHARP)}\`}
-    php={\`${fileTabContentsForLanguage(value, ExampleLanguage.PHP)}\`}
-    rust={\`${fileTabContentsForLanguage(value, ExampleLanguage.RUST)}\`}
-    ruby={\`${fileTabContentsForLanguage(value, ExampleLanguage.RUBY)}\`}
-    elixir={\`${fileTabContentsForLanguage(value, ExampleLanguage.ELIXIR)}\`}
-    swift={\`${fileTabContentsForLanguage(value, ExampleLanguage.SWIFT)}\`}
-    dart={\`${fileTabContentsForLanguage(value, ExampleLanguage.DART)}\`}
-    cli={\`${fileTabContentsForLanguage(value, ExampleLanguage.CLI)}\`}
-  />
-  `;
+  node.name = 'SdkExampleFileTabsImpl';
+  node.attributes = attributes;
 }
 
-function replaceValueWithExampleCodeBlock(literal: unist.Literal): void {
-  const value = literal.value as string;
+function replaceValueWithExampleCodeBlock(node: MdxJsxFlowElement): void {
+  const attrmap: Record<string, string | undefined> = {};
+  for (const a of node.attributes) {
+    if (a.type !== 'mdxJsxAttribute') {
+      continue;
+    }
+    const value = resolveAttributeValue(a);
+    if (value === undefined) {
+      throw Error(`Invalid undefined value for SdkExampleCodeBlock.${a.name}`);
+    }
+    attrmap[a.name] = value;
+  }
 
-  const snippetLanguageAttr = value.match(/language={'([^']+)'}/m)?.[1];
+  const snippetLanguageAttr = attrmap['language'];
   if (snippetLanguageAttr === undefined) {
-    console.log(
-      `Unable to find "language={'xxx'}" attribute on jsx tag:\n${value}`
-    );
-    return;
+    throw Error('Missing attribute SdkExampleCodeBlock.language');
   }
   const language = exampleLanguage(snippetLanguageAttr);
 
-  const snippetId = value.match(/snippetId={'([^']+)'}/m)?.[1];
-  const filename = value.match(/file={'([^']+)'}/m)?.[1];
-  if (snippetId === undefined && filename === undefined) {
-    console.log(
-      `Unable to find "snippetId={'xxx'}" or "file={'xxx'}" attribute on jsx tag:\n${value}`
```

Comment:
These log statements were intended to give us a way to audit all the code blocks in the docs and swarm to fill in missing ones, someday, when time permits. Did you have another idea in mind for how we can easily identify the list of example snippets that are missing for a given language?

Reply:
This is actually the error produced when an `SdkExample*` element is being parsed. In this specific case, it's complaining that the element is missing required attributes (either `snippetId` or `file`).

Missing snippet / file detection is still handled exactly the same way, and `yarn clean && yarn build` will dump out a report.

I generally converted `console.log()` to `throw Error()` in the code injection plugin so that we're not quietly skipping any misconfigured `SdkExample*` elements.